packages/client/src/authentication/Authentication.ts

```diff
@@ -6,12 +6,39 @@ import { CredentialsExpiredError, NotAuthenticatedError } from '../consts/errors
 import { AuthenticationApi } from './adapters/AuthenticationApi';
 import { CredentialsStorage } from './adapters/CredentialsStorage';
 
+/**
+ * Authentication for Lens API.
+ *
+ * @group LensClient Modules
+ */
 export interface IAuthentication {
+  /**
+   * Generate a challenge string for the wallet to sign.
+   *
+   * @param address - The wallet address
+   * @returns A challenge string
+   */
   generateChallenge(address: string): Promise<string>;
```

Comment:
Maybe for a future iteration. We do have a alias called `EthereumAddress` in `shared-kernel` which can help make signatures more clear. In the future we might also brand the `EthereumAddress` so that, with nominal typing derived from the branding, it's not possible to make mistakes.